sysl2/sysl/exporter/endpoint_exporter.go

```diff
@@ -130,41 +130,52 @@ func (e *EndpointExporter) setHTTPMethod(path string, endpoint *proto.Endpoint,
 	return op
 }
 
+func (e *EndpointExporter) setCommonAttributes(
+	name string,
+	param *spec.Parameter,
+	attrMap map[string]*proto.Attribute,
+	valueMap protoType,
+) {
+	param.Format = valueMap.Format
+	param.Type = valueMap.Type
+	param.Name = name
+	if _, ok := attrMap["required"]; ok {
+		param.Required = true
+	} else {
+		param = param.AsOptional()
+	}
+	if param.Type == object {
+		param.Schema.ExtraProps["$ref"] = "#/definitions/" + param.Format
+	}
+}
+
 func (e *EndpointExporter) setPathParams(endpoint *proto.Endpoint, op *spec.Operation) error {
+	var attrMap map[string]*proto.Attribute
 	for _, inParam := range endpoint.GetRestParams().GetUrlParam() {
+		attrMap = inParam.GetType().GetAttrs()
 		param := spec.PathParam(inParam.GetName())
```

Comment:
it looks like this line is now the only difference in these two loops, Can you pass `spec.PathParam`/`spec.QueryParam` as a func pointer to the setCommonAttributes() func to trim this down to just the last 2 lines?

Reply:
Obsession to DRY is bad and impacts the readability of the code.

Reply:
100% DISAGREE! when multiple sections/functions/whatever has almost the same code there is additional work needed by people coming back to the code, usually with questions like:
1. are the differences here what is actually required? Or is one buggy (and then which one?)
1. I have to fix a bug (or requirement change) in this code, how am I supposed to find out where else the same construct exists which needs the same change?

This link [pdf](https://people.cs.uchicago.edu/~shanlu/paper/TSE-CPMiner.pdf) isnt what I was looking for but its close enough, its a long article about why duplicated code sections are dangerous (from the point of view of how they were duplicated - copy + pasting code snippets), the link I was actually looking for was an article by (I think) Jon Carmack showing some of the copy+paste bugs which appeared in the ID software's game engines.

so No, DRY is absolutely important and is almost the #1 rule

Reply:
Ok may be my comment was a little awry, the disagree is in understanding what exactly is a repetition, it would be easier to discuss F2f.